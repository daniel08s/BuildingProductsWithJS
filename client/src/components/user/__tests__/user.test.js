/* global test expect shallow React */
// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import UserWrapper, {User} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '1',
  login: 'testuser',
  registrationDate: new Date(2018, 1, 1, 1, 1, 1, 1),
};

test('# UserWrapper', () => {
  const store = mockStore({auth: {user}});

  const wrapper = shallow(<UserWrapper user={user} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# User', () => {
  const newLogin = 'testuser new';
  const userUpdated = {
    id: '1',
    login: newLogin,
    registrationDate: new Date(2018, 1, 1, 1, 1, 1, 1),
  };
  const updateUser = u => expect(u).toEqual(userUpdated);
  const component = (
    <User
      user={user}
      edit
      handleUpdateUser={updateUser}
    />
  );
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // test interaction
  const app = mount(component);

  /* Test Answer */
  // set login
  app.find('#loginText').getDOMNode().value = newLogin;

  // click answer button
  app.find('#saveBtn').simulate('click');
});
