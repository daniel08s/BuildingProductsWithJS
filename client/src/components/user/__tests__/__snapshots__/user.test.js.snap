// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# User 1`] = `
<div
  className="panel panel-default text-left"
>
  <div
    className="panel-heading form-inline"
  >
    User: 
    <input
      className="form-control"
      defaultValue="testuser"
      id="loginText"
      type="text"
    />
    <div
      className="pull-right"
    >
      <button
        className="btn btn-default"
        id="saveBtn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="panel-body"
  >
    Registration date: 
    Thu Feb 01 2018 01:01:01 GMT+0000
  </div>
</div>
`;

exports[`# UserWrapper 1`] = `
<User
  edit={false}
  handleUpdateUser={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  user={
    Object {
      "id": "1",
      "login": "testuser",
      "registrationDate": 2018-02-01T01:01:01.001Z,
    }
  }
/>
`;
